$OpenBSD: patch-parser_c,v 1.1.1.1.2.1 2000/09/15 05:22:23 marc Exp $

Was a core fixer in a previous version, should still be useful with
this one.

--- parser.c.orig	Sun Apr 30 11:23:55 2000
+++ parser.c	Sat May 20 02:11:48 2000
@@ -415,6 +415,9 @@
 
 void ParseTag(Lexer *lexer, Node *node, uint mode)
 {
+    if (!node->tag)
+        return;
+
     if (node->tag->model & CM_EMPTY)
     {
         lexer->waswhite = no;
@@ -1583,7 +1586,7 @@
             continue;
         }
 
-        if (node->tag != tag_li)
+        if ((node->tag != tag_li) && node->tag)
         {
             UngetToken(lexer);
 
